internal/image/layer.go

```diff
@@ -47,7 +111,43 @@ type Layer struct {
 	// TODO: Use hashmap to speed up path lookups
 }
 
+func (filemap Layer) Open(path string) (fs.File, error) {
+	node, err := filemap.getFileNode(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return node.Open()
+}
+
+func (filemap Layer) Stat(path string) (fs.FileInfo, error) {
+	node, err := filemap.getFileNode(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return node.Stat()
+}
+
+func (filemap Layer) ReadDir(path string) ([]fs.DirEntry, error) {
+	children := filemap.fileNodeTrie.GetChildren(path)
+	output := make([]fs.DirEntry, 0, len(children))
+	for _, node := range children {
+		output = append(output, node)
+	}
+
+	return output, nil
+}
+
+var _ fs.FS = Layer{}
+var _ fs.StatFS = Layer{}
+var _ fs.ReadDirFS = Layer{}
+
 func (filemap Layer) getFileNode(path string) (*FileNode, error) {
+	if !filepath.IsAbs(path) {
```

Comment:
Can you add a comment here explaining why we need to handle this? 

As opposed to say, enforcing this precondition that paths passed have to be absolute paths?

Reply:
Added a comment:
```
	// We expect all paths queried to be absolute paths rooted at the container root
	// However, scalibr uses paths without a prepending /, because the paths are relative to ScanRoot
	// ScanRoot will always be '/' for container scanning, so prepend with / if necessary.
```